diesel/src/sqlite/connection/mod.rs

```diff
@@ -372,4 +372,102 @@ mod tests {
             .get_result::<(i32, i32, i32)>(&connection);
         assert_eq!(Ok((2, 3, 4)), added);
     }
+
+    use crate::sqlite::Aggregator;
+    use crate::sql_types::Foldable;
+    use std::ops::AddAssign;
+    sql_function! {
+        #[aggregate]
+        fn my_sum<ST: Foldable>(expr: ST) -> ST::Sum;
+    }
+
+    #[derive(Default)]
+    struct MySum<T> {
+        sum: T,
+    }
+    impl<T: Default + AddAssign> Aggregator<T> for MySum<T> {
+        type Output = T;
+
+        fn step(&mut self, expr: T) {
+            self.sum += expr;
+        }
+
+        fn finalize(self) -> Self::Output {
+            self.sum
+        }
+    }
+    table! {
+        my_sum_example {
+            id -> Integer,
+            value -> Integer,
+        }
+    }
+
+    #[test]
+    fn register_aggregate_function() {
+        use self::my_sum_example::dsl::*;
+
+        let connection = SqliteConnection::establish(":memory:").unwrap();
+        connection.execute("CREATE TABLE my_sum_example (id integer primary key autoincrement, value integer)").unwrap();
+        connection.execute("INSERT INTO my_sum_example (value) VALUES (1), (2), (3)").unwrap();
+
+        my_sum::register_impl::<MySum>(&connection).unwrap();
+
+        let result = crate::select(my_sum(value)).get_result::<i32>(&connection);
+        assert_eq!(Ok(6), result);
+    }
+
+    use crate::sql_types::SqlOrd;
+    sql_function! {
+        #[aggregate]
+        fn range_max<T: SqlOrd>(exprs: (T, T, T)) -> Option<T>;
+    }
+
+    #[derive(Default)]
+    struct RangeMax<T: Default> {
+        max_value: Option<T>,
+    }
+    impl<T: Default + Ord> Aggregator<(T, T, T)> for RangeMax<T> {
+        type Output = Option<T>;
+
+        fn step(&mut self, args: (T, T, T)) {
+            let (x0, x1, x2) = args;
```

Comment:
Just a small nit: This can also be written as
```suggestion
        fn step(&mut self, (x0, x1, x2): (T, T, T)) {
```

Reply:
Oh right! I will make sure to add that in my next commit